components/node-labeler/cmd/run.go

```diff
@@ -110,8 +114,40 @@ var runCmd = &cobra.Command{
 			log.WithError(err).Fatal("unable to bind controller watch event handler")
 		}
 
+		if err := mgr.GetFieldIndexer().IndexField(context.Background(), &workspacev1.Workspace{}, "status.runtime.nodeName", func(o client.Object) []string {
```

Comment:
@filiptronicek do you recall for what purpose the indexer was added?

As an example you can see here, where Alejandro added one to avoid metric duplication.

https://github.com/gitpod-io/gitpod/blob/eb3a811708131c8d94ad37fe0bf5c24748fd8c97/components/ws-manager-mk2/controllers/workspace_controller.go#L650  (the PR is https://github.com/gitpod-io/gitpod/pull/18625).

I wonder if maybe it can be removed for now.

Reply:
This is in regards to L117-L126

Reply:
We need the indexer, because we rely on it here: https://github.com/gitpod-io/gitpod/blob/eb3a811708131c8d94ad37fe0bf5c24748fd8c97/components/node-labeler/cmd/run.go#L447

Synced with Filip for ☝️